flink-connectors/flink-connector-rabbitmq/src/test/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSourceTest.java

```diff
@@ -318,6 +320,36 @@ public void testConstructorParams() throws Exception {
 		assertEquals("passTest", testObj.getFactory().getPassword());
 	}
 
+	/**
+	 * Tests getting the correct body and correlationID given which constructor was called.
+	 * if the constructor with the {@link DeserializationSchema} was called it should extract the body of the message
+	 * from the {@link Delivery} and the correlation ID from the {@link AMQP.BasicProperties} which are
+	 * mocked to "I Love Turtles" and "0".
+	 * if the constructor with the {@link RMQDeserializationSchema} was called it uses the
+	 * {@link RMQDeserializationSchema#deserialize} method to parse the message and extract the correlation ID which
+	 * both are implemented in {@link RMQTestSource#initAMQPMocks()} to return the
+	 * {@link AMQP.BasicProperties#getMessageId()} that is mocked to return "1-MESSAGE_ID"
+	 */
+	@Test
+	public void testProcessMessage() throws Exception {
```

Comment:
Please fix all warnings in this test. There are a few warnings about using raw types.

Reply:
Fixed all warnings but the unchecked generics. apparently that's a known issue: https://github.com/mockito/mockito/issues/1531

Reply:
Now without the warnings I took a closer look at the test and I thought maybe we could improve it a bit? I do prefer blackbox kind of tests. What do you think about a test like follows:

```
	@Test(timeout = 10000L)
	public void testCustomIdentifiers() throws Exception {
		source = new RMQTestSource(new CustomDeserializationSchema());
		sourceThread.start();
		source.initializeState(getMockContext());
		source.open(config);
		sourceThread.join();

		assertThat(DummySourceContext.numElementsCollected, equalTo(2L));
	}

	private static class CustomDeserializationSchema implements RMQDeserializationSchema<String> {
		@Override
		public TypeInformation<String> getProducedType() {
			return TypeExtractor.getForClass(String.class);
		}

		@Override
		public void deserialize(Envelope envelope, AMQP.BasicProperties properties, byte[] body, RMQCollector<String> collector){
			String correlationId = properties.getCorrelationId();
			if (correlationId.equals("1")) {
				collector.setMessageIdentifiers("1", envelope.getDeliveryTag());
				collector.collect("I Love Turtles");
				collector.collect("Brush your teeth");
			} else if (correlationId.equals("2")) {
				// should not be emitted, because it has the same correlationId as the previous one
				collector.setMessageIdentifiers("1", envelope.getDeliveryTag());
				collector.collect("Brush your teeth");
			} else {
				// should end the stream, should not be emitted
				collector.setMessageIdentifiers("2", envelope.getDeliveryTag());
				collector.collect("FINISH");
			}
		}

		@Override
		public boolean isEndOfStream(String record) {
			return record.equals("FINISH");
		}
	}
```